tests/unit/hive_metastore/test_locations.py

```diff
@@ -734,3 +735,188 @@ def test_resolve_dbfs_root_in_hms_federation():
+    backend = MockBackend(
+        rows={
+            '`hive_metastore`.`test`.`tables`': [],
+            '`test`.`mounts`': MOUNT_STORAGE[("/mnt/test_mount", "adls://bucket/")],
+        }
+    )
+    mounts = MountsCrawler(backend, client, "test")
+    results = TablesInMounts(backend, client, "test", mounts).snapshot(force_refresh=True)
+    assert results == [
+        Table(
+            "hive_metastore",
+            "mounted_test_mount",
+            "table1",
+            "EXTERNAL",
+            "JSON",
+            "adls://bucket/entity/domain/table1",
+        ),
+        Table(
+            "hive_metastore",
+            "mounted_test_mount",
+            "domain",
+            "EXTERNAL",
+            "CSV",
+            "adls://bucket/entity/domain",
+        ),
+        Table(
+            "hive_metastore",
+            "mounted_test_mount",
+            "z_dir",
+            "EXTERNAL",
+            "JSON",
+            "adls://bucket/entity/domain/z_dir",
+        ),
+    ]
+
+
+def test_mount_dont_list_partitions():
+    client = create_autospec(WorkspaceClient)
+
+    first_folder = FileInfo("dbfs:/mnt/test_mount/entity/", "entity/", 0, "")
+    first_first_partition = FileInfo("dbfs:/mnt/test_mount/entity/xxx=yyy/", "xxx=yyy/", 0, "")
+    first_first_partition_files = FileInfo("dbfs:/mnt/test_mount/entity/xxx=yyy/1.parquet", "1.parquet", "", "")
+    first_second_partition = FileInfo("dbfs:/mnt/test_mount/entity/xxx=zzz/", "xxx=zzz/", 0, "")
+    first_second_partition_files = FileInfo("dbfs:/mnt/test_mount/entity/xxx=zzz/1.parquet", "1.parquet", "", "")
+    misplaced_json = FileInfo("dbfs:/mnt/test_mount/entity/xxx=zzz/misplaced.json", "misplaced.json", "", "")
+    first_delta_log = FileInfo("dbfs:/mnt/test_mount/entity/_delta_log/", "_delta_log/", 0, "")
+
+    def my_side_effect(path, **_):
+        if path == "/mnt/test_mount":
+            return [first_folder]
+        if path == "dbfs:/mnt/test_mount/entity/":
+            return [first_delta_log, first_first_partition, first_second_partition]
+        if path == "dbfs:/mnt/test_mount/entity/xxx=yyy/":
+            return [first_first_partition_files, misplaced_json]
+        if path == "dbfs:/mnt/test_mount/entity/xxx=zzz/":
+            return [first_second_partition_files]
+        return None
+
+    client.dbutils.fs.ls.side_effect = my_side_effect
+    backend = MockBackend(
+        rows={
+            '`hive_metastore`.`test`.`tables`': [],
+            '`test`.`mounts`': MOUNT_STORAGE[("/mnt/test_mount", "adls://bucket/")],
+        }
+    )
+    mounts = MountsCrawler(backend, client, "test")
+    results = TablesInMounts(backend, client, "test", mounts).snapshot(force_refresh=True)
+    assert len(results) == 1
+    assert results[0].table_format == "DELTA"
+    assert results[0].is_partitioned
+
+
+@pytest.fixture
+def set_recursion_limit():
+    original_limit = sys.getrecursionlimit()
+    sys.setrecursionlimit(150)
```

Comment:
please rewrite algorithm using the iterative approach, e.g. managing stack via list. if this hack is required on the testing side, it'll blow up in the customer side.

Reply:
This was just a test to catch an infinite loop in the code, which was a bug I had to solve when I picked up this code initially. I can get rid of the test if it's causing too much confusion.